### PR TITLE
DIV-5251: Redirect Respondent to DA if divorce is granted

### PR DESCRIPTION
**Pull request by johnoneill46**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: middleware/petitionMiddleware.js [hunk 1]
```diff
@@ -60,6 +62,10 @@ const loadMiniPetition = (req, res, next) => {
         if (idamUserIsCorespondent) {
           return res.redirect(findCoRespPath(coRespAnswers, caseState));
```
Comment (yarpo) [inline middleware/petitionMiddleware.js]: I know it's old code, but I can see we have logs everywhere else, but not here. Is that ok?

Can you please add log here as well if there is not reason why we shouldn't add it here.

File: middleware/redirectMiddleware.js [hunk 2]
```diff
@@ -23,4 +24,12 @@ const redirectOnCondition = (req, res, next) => {
   return next();
 };
 
-module.exports = { redirectOnCondition };
+const redirectToDa = (req, res) => {
+  const DaAppLandingPage = `${config.services.daFrontend.url}${config.services.daFrontend.landing}`;
```
Comment (yarpo) [inline middleware/redirectMiddleware.js]: shouldn't var name start with lower case?

File: middleware/redirectMiddleware.js [hunk 3]
```diff
@@ -23,4 +24,12 @@ const redirectOnCondition = (req, res, next) => {
...
+  const DaQueryString = `?${authTokenString}=${req.cookies[authTokenString]}`;
```
Comment (yarpo) [inline middleware/redirectMiddleware.js]: as above

**Comment by qzhou-hmcts** [comment]
I see you guys are using DA app for both petitioner and respondent users. I wondering how are you handling to retrieve the case and differentiate between each role

File: middleware/petitionMiddleware.js (hunk 1 above)
Comment (johnoneill46) [reply middleware/petitionMiddleware.js]: Resolving as changes made

File: infrastructure/variables.tf [hunk 4]
```diff
@@ -122,13 +122,16 @@ variable "idam_authentication_login_endpoint" {
...
+}
 
 variable "decree_nisi_frontend_url" {
   type = "string"
 }
 
-variable "frontend_service_name" {
-  default = "divorce-respondent-frontend"
+variable "decree_absolute_frontend_url" {
+  default = ""
```
Comment (tchow8) [inline infrastructure/variables.tf]: You can type this and remove the default, to enforce that it must be defined.

File: middleware/petitionMiddleware.js [hunk 5]
```diff
@@ -48,6 +50,7 @@ function findCoRespPath(coRespAnswers, caseState) {
 
 const loadMiniPetition = (req, res, next) => {
   return caseOrchestration.getPetition(req)
+  // eslint-disable-next-line complexity
```
Comment (tchow8) [inline middleware/petitionMiddleware.js]: Probably better to split into separate code blocks rather than disable.

**Comment by nsingamsetty** [comment]
This is tested and signed off

**Pull request merged**